tests/Unit/PrestaShopBundle/Translation/Loader/LegacyFileLoaderTest.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Unit\PrestaShopBundle\Translation\Loader;
+
+use PrestaShopBundle\Translation\Loader\LegacyFileLoader;
+use PrestaShopBundle\Translation\Loader\LegacyFileReader;
+use Symfony\Component\Translation\MessageCatalogue;
+
+class LegacyFileLoaderTest extends \PHPUnit_Framework_TestCase
```

Comment:
already tested here => https://github.com/PrestaShop/PrestaShop/blob/develop/tests/Integration/PrestaShopBundle/Translation/Loader/LegacyFileLoaderTest.php

Or is it a new LegacyFileLoader?

Reply:
There are two tests for this class, a unit test and an integration test. The unit test provides a mock reader with hardcoded data. The integration test tests the actual service which is configured with the real reader that reads data from real files.